includes/class-wc-stripe-blocks-support.php

```diff
@@ -0,0 +1,345 @@
+<?php
+use Automattic\WooCommerce\Blocks\Payments\Integrations\AbstractPaymentMethodType;
+use Automattic\WooCommerce\Blocks\Payments\PaymentResult;
+use Automattic\WooCommerce\Blocks\Payments\PaymentContext;
+use Automattic\WooCommerce\Blocks\Payments\Api as PaymentMethodsAPI;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Stripe_Blocks_Support class.
+ *
+ * @extends AbstractPaymentMethodType
+ */
+final class WC_Stripe_Blocks_Support extends AbstractPaymentMethodType {
+	/**
+	 * Payment method name defined by payment methods extending this class.
+	 *
+	 * @var string
+	 */
+	protected $name = 'stripe';
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_rest_checkout_process_payment_with_context', [ $this, 'add_payment_request_order_meta' ], 8, 2 );
+		add_action( 'woocommerce_rest_checkout_process_payment_with_context', [ $this, 'add_stripe_intents' ], 9999, 2 );
+	}
+
+	/**
+	 * Initializes the payment method type.
+	 */
+	public function initialize() {
+		$this->settings = get_option( 'woocommerce_stripe_settings', [] );
+	}
+
+	/**
+	 * Returns if this payment method should be active. If false, the scripts will not be enqueued.
+	 *
+	 * @return boolean
+	 */
+	public function is_active() {
+		return ! empty( $this->settings['enabled'] ) && 'yes' === $this->settings['enabled'];
+	}
+
+	/**
+	 * Returns an array of scripts/handles to be registered for this payment method.
+	 *
+	 * @return array
+	 */
+	public function get_payment_method_script_handles() {
+		$asset_path = WC_STRIPE_PLUGIN_PATH . '/build/index.asset.php';
+		$version = WC_STRIPE_VERSION;
+		$dependencies = [];
+		if ( file_exists( $asset_path ) ) {
+			$asset = require $asset_path;
+			$version = is_array( $asset ) && isset( $asset['version'] )
+				? $asset['version']
+				: $version;
+			$dependencies = is_array( $asset ) && isset( $asset['dependencies'] )
+				? $asset['dependencies']
+				: $dependencies;
+		}
+		wp_register_script(
+			'wc-stripe-blocks-integration',
+			WC_STRIPE_PLUGIN_URL . '/build/index.js',
+			$dependencies,
+			$version,
+			true
+		);
+
```

Comment:
I forgot to add `wp_set_script_translations` call here for registering that the stripe js has translation strings. Is that something you folks can take care of?

Reply:
Yep, we'll take care of that! Thanks for pointing it out ☺️

Reply:
@nerrad could you please confirm that the way I did this in e1b9fedf46337759e88a137c16dab2587e7289a5 is the right way?

I couldn't find any prior art in how the function is used 🤔

Reply:
Looks good to me! Given Stripe is published to wp.org the JS translations will be served from the language packs on wp.org with the current implementation of the function.